### ISSUE TITLE
DynamoDB table-class Parameter (DynamoDB Standard-IA)

### ISSUE THREAD

**Issue by AdamTylerLynch** [hashicorp/terraform-provider-aws#22141]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Amazon DynamoDB Standard-Infrequent Access (DynamoDB Standard-IA). A new table class for DynamoDB that reduces storage costs by 60 percent compared to existing DynamoDB Standard tables, and that delivers the same performance, durability, and scaling.

### New or Affected Resource(s)

* aws_dynamodb_table

### Potential Terraform Configuration

```hcl

resource "aws_dynamodb_table" "basic-dynamodb-table" {
  name           = "GameScores"
  billing_mode   = "PROVISIONED"
  read_capacity  = 20
  write_capacity = 20
  hash_key       = "UserId"
  range_key      = "GameTitle"
  table_class    = "STANDARD_INFREQUENT_ACCESS"

  attribute {
    name = "UserId"
    type = "S"
  }

  attribute {
    name = "GameTitle"
    type = "S"
  }

  attribute {
    name = "TopScore"
    type = "N"
  }

  ttl {
    attribute_name = "TimeToExist"
    enabled        = false
  }

  global_secondary_index {
    name               = "GameTitleIndex"
    hash_key           = "GameTitle"
    range_key          = "TopScore"
    write_capacity     = 10
    read_capacity      = 10
    projection_type    = "INCLUDE"
    non_key_attributes = ["UserId"]
  }

  tags = {
    Name        = "dynamodb-table-1"
    Environment = "production"
  }
}
```

### References

* https://aws.amazon.com/blogs/aws/new-dynamodb-table-class-save-up-to-60-in-your-dynamodb-costs/

**Comment by AdamTylerLynch** [comment]
Include ReplicationGroups to validate transient properties of table_class.

**Comment by ewbankkit** [comment]
Relates: https://github.com/hashicorp/terraform-provider-aws/issues/21995.
Relates: https://github.com/hashicorp/terraform-provider-aws/pull/22110.

**Comment by AdamTylerLynch** [comment]
Duplicated: #21995
Closed by: #22110
